pkg/webui/console/components/user-data-form/index.js

```diff
@@ -79,53 +74,14 @@ const validationSchema = Yup.object().shape({
   description: Yup.string().max(2000, Yup.passValues(sharedMessages.validateTooLong)),
 })
 
-const createPasswordValidationSchema = requirements => {
-  const passwordValidation = Yup.string()
-    .required(sharedMessages.validateRequired)
-    .test(
-      'min-length',
-      { message: sharedMessages.validateTooShort, values: { min: requirements.min_length } },
-      password => hasMinLength(password, requirements.min_length),
-    )
-    .test(
-      'max-length',
-      { message: sharedMessages.validateTooLong, values: { max: requirements.max_length } },
-      password => hasMaxLength(password, requirements.max_length),
-    )
-    .test(
-      'min-special',
-      { message: m.validateSpecial, values: { special: requirements.min_special } },
-      password => hasSpecial(password, requirements.min_special),
-    )
-    .test(
-      'min-upper',
-      { message: m.validateUppercase, values: { upper: requirements.min_uppercase } },
-      password => hasUpper(password, requirements.min_uppercase),
-    )
-    .test(
-      'min-digit',
-      { message: m.validateDigit, values: { digit: requirements.min_digits } },
-      password => hasDigit(password, requirements.min_digits),
-    )
-
-  return validationSchema.concat(
-    Yup.object().shape({
-      password: passwordValidation,
-      confirmPassword: Yup.string()
-        .required(sharedMessages.validateRequired)
-        .oneOf([Yup.ref('password'), null], sharedMessages.validatePasswordMatch),
-    }),
-  )
-}
-
 @injectIntl
 class UserForm extends React.Component {
   constructor(props) {
     super(props)
 
     const { update, passwordRequirements } = props
     this.validationSchema = update
-      ? validationSchema
+      ? baseValidationSchema
       : createPasswordValidationSchema(passwordRequirements)
```

Comment:
We need to merge password validation with validation rules for other fields as well. With this implementation:
1. Form values are no properly casted
2. Multiple fields are missing validation